frontend/alert/src/components/AutoComplete.js

```diff
@@ -0,0 +1,237 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import AwesomeDebouncePromise from "awesome-debounce-promise";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHistory } from "@fortawesome/free-solid-svg-icons";
+import { useOnClickOutside } from "./shared/useOnClickOutside";
+import { Input } from "./Input";
+
+/* A function that takes in a search term and returns a promise with both the search term and
+the search results.
+Including the search term makes it possible to determine if the search result is stale.
+ */
+const suggestionsFor = search => fetch(`/api/alert/courses?search=${search}`)
+    .then(res => res.json()
+        .then(searchResult => ({
+            searchResult,
+            searchTerm: search,
+        })));
+
+/* Debounce the search promise so that it doesn't make requests to the backend more frequently
+than the given interval
+*/
+const SUGGESTION_INTERVAL = 250;
+const suggestionsDebounced = AwesomeDebouncePromise(
+    suggestionsFor,
+    SUGGESTION_INTERVAL,
+);
+
+const AUTOCOMPLETE_BORDER_WIDTH = 1;
+
+const SuggestionsContainer = styled.div`
+    position: absolute;
+    left: 0;
+    top: 100%;
+    width: ${({ below }) => below && (below.getBoundingClientRect().width - AUTOCOMPLETE_BORDER_WIDTH * 2)}px;
+    visibility: ${({ hidden }) => (hidden ? "hidden" : "visible")};
+    z-index:5000;
+    text-align: left;
+`;
+
+const SuggestionsBox = styled.div`
+    padding-bottom: 0.5rem;
+    border-color: #d6d6d6;
+    border-width: ${AUTOCOMPLETE_BORDER_WIDTH}px;
+    margin-top: 0.7rem;
+    margin-right: 0;
+    width: 100%;
+    border-style: solid;
+    background-color: white;
+    padding-top: 0;
+    max-height: 20rem;
+    overflow-y: scroll;
+    &::-webkit-scrollbar {
+        display: none;
+    }
+`;
+
+const SuggestionBox = styled.div`
```

Comment:
These two components have kind of confusing names, also along with `SuggestionsContainer` and a few others. Not sure if there's a way around this -- since styled components can make that funky.